mobile/lib/ui/viewer/actions/file_selection_actions_widget.dart

```diff
@@ -764,8 +811,21 @@ class _FileSelectionActionsWidgetState
       );
       final String url =
           "${_cachedCollectionForSharedLink!.publicURLs?.first?.url}#$collectionKey";
+      await shareImageAndUrl(generatedPathNotifier.value!, url);
       await Clipboard.setData(ClipboardData(text: url));
       showShortToast(context, S.of(context).linkCopiedToClipboard);
+      if (generatedPathNotifier.value != null) {
+        final file = File(generatedPathNotifier.value!);
+        try {
+          if (await file.exists()) {
+            await file.delete();
```

Comment:
Use `existsSync()` and `deleteSync()`. 
Ref: https://stackoverflow.com/questions/61169619/dartio-sync-vs-async-file-operations